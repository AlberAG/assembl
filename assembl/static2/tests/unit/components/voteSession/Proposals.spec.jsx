import React from 'react';
import renderer from 'react-test-renderer';

import Proposals from '../../../../js/app/components/voteSession/proposals';
import * as fakeData from './fakeData';

describe('Proposals component', () => {
  it('should match Proposals snapshot', () => {
    const { proposals, remainingTokensByCategory, userGaugeVotes, userTokenVotes } = fakeData;
    const props = {
      proposals: proposals,
      remainingTokensByCategory: remainingTokensByCategory,
      userGaugeVotes: userGaugeVotes,
      userTokenVotes: userTokenVotes
    };
    const rendered = renderer.create(<Proposals {...props} />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});