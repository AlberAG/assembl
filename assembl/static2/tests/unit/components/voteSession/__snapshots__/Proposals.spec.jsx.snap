// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Proposals component should match Proposals snapshot 1`] = `
<div
  className="container"
>
  <div
    className="theme-box"
  >
    <div
      className="proposal row"
    >
      <div
        className="col-md-6 col-xs-6"
      >
        <h3
          className="dark-title-3"
        >
          Foo
        </h3>
        <p>
          You can't hack the alarm without connecting the primary AGP microchip!
        </p>
      </div>
      <div
        className="col-md-6 col-xs-6"
      >
        <div>
          If we generate the feed, we can get to the AGP driver through the primary ADP interface!
          <div>
            <p>
              Positive
            </p>
            <div
              className="tokens"
            >
              <button
                className="admin-icons btn btn-default"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                <svg
                  aria-hidden="true"
                  height="35px"
                  role="img"
                  version="1.1"
                  viewBox="0 0 20 20"
                  width="35px"
                  x="0px"
                  xmlSpace="preserve"
                  y="0px"
                >
                  <path
                    className="outer"
                    d="m 15.59322,11.271187 a 6.2288136,6.3559322 0 1 1 -12.4576271,0 6.2288136,6.3559322 0 1 1 12.4576271,0 z"
                    fill="#226622"
                    id="path2999"
                    stroke="#226622"
                    strokeDasharray="none"
                    strokeMiterlimit="4"
                    strokeOpacity="1"
                    strokeWidth="1"
                    transform="matrix(1.4872568,0,0,1.4575117,-3.9272774,-6.4278868)"
                  />
                </svg>
              </button>
              <button
                className="admin-icons btn btn-default"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                <svg
                  aria-hidden="true"
                  height="35px"
                  role="img"
                  version="1.1"
                  viewBox="0 0 20 20"
                  width="35px"
                  x="0px"
                  xmlSpace="preserve"
                  y="0px"
                >
                  <path
                    className="outer"
                    d="m 15.59322,11.271187 a 6.2288136,6.3559322 0 1 1 -12.4576271,0 6.2288136,6.3559322 0 1 1 12.4576271,0 z"
                    fill="#226622"
                    id="path2999"
                    stroke="#226622"
                    strokeDasharray="none"
                    strokeMiterlimit="4"
                    strokeOpacity="1"
                    strokeWidth="1"
                    transform="matrix(1.4872568,0,0,1.4575117,-3.9272774,-6.4278868)"
                  />
                </svg>
              </button>
              <button
                className="admin-icons btn btn-default"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                <svg
                  aria-hidden="true"
                  height="35px"
                  role="img"
                  version="1.1"
                  viewBox="0 0 20 20"
                  width="35px"
                  x="0px"
                  xmlSpace="preserve"
                  y="0px"
                >
                  <path
                    className="outer"
                    d="m 15.59322,11.271187 a 6.2288136,6.3559322 0 1 1 -12.4576271,0 6.2288136,6.3559322 0 1 1 12.4576271,0 z"
                    fill="none"
                    id="path2999"
                    stroke="#226622"
                    strokeDasharray="none"
                    strokeMiterlimit="4"
                    strokeOpacity="1"
                    strokeWidth="1"
                    transform="matrix(1.4872568,0,0,1.4575117,-3.9272774,-6.4278868)"
                  />
                </svg>
              </button>
              <button
                className="admin-icons btn btn-default"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                <svg
                  aria-hidden="true"
                  height="35px"
                  role="img"
                  version="1.1"
                  viewBox="0 0 20 20"
                  width="35px"
                  x="0px"
                  xmlSpace="preserve"
                  y="0px"
                >
                  <path
                    className="outer"
                    d="m 15.59322,11.271187 a 6.2288136,6.3559322 0 1 1 -12.4576271,0 6.2288136,6.3559322 0 1 1 12.4576271,0 z"
                    fill="none"
                    id="path2999"
                    stroke="#226622"
                    strokeDasharray="none"
                    strokeMiterlimit="4"
                    strokeOpacity="1"
                    strokeWidth="1"
                    transform="matrix(1.4872568,0,0,1.4575117,-3.9272774,-6.4278868)"
                  />
                </svg>
              </button>
              <button
                className="admin-icons btn btn-default"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                <svg
                  aria-hidden="true"
                  height="35px"
                  role="img"
                  version="1.1"
                  viewBox="0 0 20 20"
                  width="35px"
                  x="0px"
                  xmlSpace="preserve"
                  y="0px"
                >
                  <path
                    className="outer"
                    d="m 15.59322,11.271187 a 6.2288136,6.3559322 0 1 1 -12.4576271,0 6.2288136,6.3559322 0 1 1 12.4576271,0 z"
                    fill="none"
                    id="path2999"
                    stroke="#226622"
                    strokeDasharray="none"
                    strokeMiterlimit="4"
                    strokeOpacity="1"
                    strokeWidth="1"
                    transform="matrix(1.4872568,0,0,1.4575117,-3.9272774,-6.4278868)"
                  />
                </svg>
              </button>
              <button
                className="btn btn-default"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseOut={[Function]}
                onMouseOver={[Function]}
                type="button"
              >
                <span
                  className="assembl-icon-delete grey"
                />
              </button>
            </div>
          </div>
          <div>
            <p>
              Negative
            </p>
            <div
              className="tokens"
            >
              <button
                className="admin-icons btn btn-default"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                <svg
                  aria-hidden="true"
                  height="35px"
                  role="img"
                  version="1.1"
                  viewBox="0 0 20 20"
                  width="35px"
                  x="0px"
                  xmlSpace="preserve"
                  y="0px"
                >
                  <path
                    className="outer"
                    d="m 15.59322,11.271187 a 6.2288136,6.3559322 0 1 1 -12.4576271,0 6.2288136,6.3559322 0 1 1 12.4576271,0 z"
                    fill="#771122"
                    id="path2999"
                    stroke="#771122"
                    strokeDasharray="none"
                    strokeMiterlimit="4"
                    strokeOpacity="1"
                    strokeWidth="1"
                    transform="matrix(1.4872568,0,0,1.4575117,-3.9272774,-6.4278868)"
                  />
                </svg>
              </button>
              <button
                className="admin-icons btn btn-default"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                <svg
                  aria-hidden="true"
                  height="35px"
                  role="img"
                  version="1.1"
                  viewBox="0 0 20 20"
                  width="35px"
                  x="0px"
                  xmlSpace="preserve"
                  y="0px"
                >
                  <path
                    className="outer"
                    d="m 15.59322,11.271187 a 6.2288136,6.3559322 0 1 1 -12.4576271,0 6.2288136,6.3559322 0 1 1 12.4576271,0 z"
                    fill="none"
                    id="path2999"
                    stroke="#771122"
                    strokeDasharray="none"
                    strokeMiterlimit="4"
                    strokeOpacity="1"
                    strokeWidth="1"
                    transform="matrix(1.4872568,0,0,1.4575117,-3.9272774,-6.4278868)"
                  />
                </svg>
              </button>
              <button
                className="admin-icons btn btn-default"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                <svg
                  aria-hidden="true"
                  height="35px"
                  role="img"
                  version="1.1"
                  viewBox="0 0 20 20"
                  width="35px"
                  x="0px"
                  xmlSpace="preserve"
                  y="0px"
                >
                  <path
                    className="outer"
                    d="m 15.59322,11.271187 a 6.2288136,6.3559322 0 1 1 -12.4576271,0 6.2288136,6.3559322 0 1 1 12.4576271,0 z"
                    fill="none"
                    id="path2999"
                    stroke="#771122"
                    strokeDasharray="none"
                    strokeMiterlimit="4"
                    strokeOpacity="1"
                    strokeWidth="1"
                    transform="matrix(1.4872568,0,0,1.4575117,-3.9272774,-6.4278868)"
                  />
                </svg>
              </button>
              <button
                className="admin-icons btn btn-default"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                <svg
                  aria-hidden="true"
                  height="35px"
                  role="img"
                  version="1.1"
                  viewBox="0 0 20 20"
                  width="35px"
                  x="0px"
                  xmlSpace="preserve"
                  y="0px"
                >
                  <path
                    className="outer"
                    d="m 15.59322,11.271187 a 6.2288136,6.3559322 0 1 1 -12.4576271,0 6.2288136,6.3559322 0 1 1 12.4576271,0 z"
                    fill="none"
                    id="path2999"
                    stroke="#771122"
                    strokeDasharray="none"
                    strokeMiterlimit="4"
                    strokeOpacity="1"
                    strokeWidth="1"
                    transform="matrix(1.4872568,0,0,1.4575117,-3.9272774,-6.4278868)"
                  />
                </svg>
              </button>
              <button
                className="btn btn-default"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseOut={[Function]}
                onMouseOver={[Function]}
                type="button"
              >
                <span
                  className="assembl-icon-delete grey"
                />
              </button>
            </div>
          </div>
        </div>
        <div
          className="gauge-vote-for-proposal"
        >
          <p>
            I'll generate the mobile SMS bus, that should microchip the TCP matrix!
          </p>
          <div
            className="rc-slider"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onMouseUp={[Function]}
            onTouchStart={[Function]}
            style={undefined}
          >
            <div
              className="rc-slider-rail"
              style={Object {}}
            />
            <div
              className="rc-slider-track"
              style={
                Object {
                  "left": "0%",
                  "visibility": "hidden",
                  "width": "NaN%",
                }
              }
            />
            <div
              className="rc-slider-step"
            />
            <div
              className="rc-slider-handle"
              role="slider"
              style={
                Object {
                  "left": "NaN%",
                }
              }
              tabIndex={0}
            />
            <div
              className="rc-slider-mark"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="theme-box"
  >
    <div
      className="proposal row"
    >
      <div
        className="col-md-6 col-xs-6"
      >
        <h3
          className="dark-title-3"
        >
          Bar
        </h3>
        <p>
          I'll input the multi-byte SAS monitor, that should bandwidth the USB microchip!
        </p>
      </div>
      <div
        className="col-md-6 col-xs-6"
      >
        <div>
          If we generate the feed, we can get to the AGP driver through the primary ADP interface!
          <div>
            <p>
              Positive
            </p>
            <div
              className="tokens"
            >
              <button
                className="admin-icons btn btn-default"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                <svg
                  aria-hidden="true"
                  height="35px"
                  role="img"
                  version="1.1"
                  viewBox="0 0 20 20"
                  width="35px"
                  x="0px"
                  xmlSpace="preserve"
                  y="0px"
                >
                  <path
                    className="outer"
                    d="m 15.59322,11.271187 a 6.2288136,6.3559322 0 1 1 -12.4576271,0 6.2288136,6.3559322 0 1 1 12.4576271,0 z"
                    fill="none"
                    id="path2999"
                    stroke="#226622"
                    strokeDasharray="none"
                    strokeMiterlimit="4"
                    strokeOpacity="1"
                    strokeWidth="1"
                    transform="matrix(1.4872568,0,0,1.4575117,-3.9272774,-6.4278868)"
                  />
                </svg>
              </button>
              <button
                className="admin-icons btn btn-default"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                <svg
                  aria-hidden="true"
                  height="35px"
                  role="img"
                  version="1.1"
                  viewBox="0 0 20 20"
                  width="35px"
                  x="0px"
                  xmlSpace="preserve"
                  y="0px"
                >
                  <path
                    className="outer"
                    d="m 15.59322,11.271187 a 6.2288136,6.3559322 0 1 1 -12.4576271,0 6.2288136,6.3559322 0 1 1 12.4576271,0 z"
                    fill="none"
                    id="path2999"
                    stroke="#226622"
                    strokeDasharray="none"
                    strokeMiterlimit="4"
                    strokeOpacity="1"
                    strokeWidth="1"
                    transform="matrix(1.4872568,0,0,1.4575117,-3.9272774,-6.4278868)"
                  />
                </svg>
              </button>
              <button
                className="admin-icons btn btn-default"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                <svg
                  aria-hidden="true"
                  height="35px"
                  role="img"
                  version="1.1"
                  viewBox="0 0 20 20"
                  width="35px"
                  x="0px"
                  xmlSpace="preserve"
                  y="0px"
                >
                  <path
                    className="outer"
                    d="m 15.59322,11.271187 a 6.2288136,6.3559322 0 1 1 -12.4576271,0 6.2288136,6.3559322 0 1 1 12.4576271,0 z"
                    fill="none"
                    id="path2999"
                    stroke="#226622"
                    strokeDasharray="none"
                    strokeMiterlimit="4"
                    strokeOpacity="1"
                    strokeWidth="1"
                    transform="matrix(1.4872568,0,0,1.4575117,-3.9272774,-6.4278868)"
                  />
                </svg>
              </button>
              <button
                className="admin-icons btn btn-default"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                <svg
                  aria-hidden="true"
                  height="35px"
                  role="img"
                  version="1.1"
                  viewBox="0 0 20 20"
                  width="35px"
                  x="0px"
                  xmlSpace="preserve"
                  y="0px"
                >
                  <path
                    className="outer"
                    d="m 15.59322,11.271187 a 6.2288136,6.3559322 0 1 1 -12.4576271,0 6.2288136,6.3559322 0 1 1 12.4576271,0 z"
                    fill="none"
                    id="path2999"
                    stroke="#226622"
                    strokeDasharray="none"
                    strokeMiterlimit="4"
                    strokeOpacity="1"
                    strokeWidth="1"
                    transform="matrix(1.4872568,0,0,1.4575117,-3.9272774,-6.4278868)"
                  />
                </svg>
              </button>
              <button
                className="admin-icons btn btn-default"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                <svg
                  aria-hidden="true"
                  height="35px"
                  role="img"
                  version="1.1"
                  viewBox="0 0 20 20"
                  width="35px"
                  x="0px"
                  xmlSpace="preserve"
                  y="0px"
                >
                  <path
                    className="outer"
                    d="m 15.59322,11.271187 a 6.2288136,6.3559322 0 1 1 -12.4576271,0 6.2288136,6.3559322 0 1 1 12.4576271,0 z"
                    fill="none"
                    id="path2999"
                    stroke="#226622"
                    strokeDasharray="none"
                    strokeMiterlimit="4"
                    strokeOpacity="1"
                    strokeWidth="1"
                    transform="matrix(1.4872568,0,0,1.4575117,-3.9272774,-6.4278868)"
                  />
                </svg>
              </button>
              <button
                className="btn btn-default"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseOut={[Function]}
                onMouseOver={[Function]}
                type="button"
              >
                <span
                  className="assembl-icon-delete grey"
                />
              </button>
            </div>
          </div>
          <div>
            <p>
              Negative
            </p>
            <div
              className="tokens"
            >
              <button
                className="admin-icons btn btn-default"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                <svg
                  aria-hidden="true"
                  height="35px"
                  role="img"
                  version="1.1"
                  viewBox="0 0 20 20"
                  width="35px"
                  x="0px"
                  xmlSpace="preserve"
                  y="0px"
                >
                  <path
                    className="outer"
                    d="m 15.59322,11.271187 a 6.2288136,6.3559322 0 1 1 -12.4576271,0 6.2288136,6.3559322 0 1 1 12.4576271,0 z"
                    fill="none"
                    id="path2999"
                    stroke="#771122"
                    strokeDasharray="none"
                    strokeMiterlimit="4"
                    strokeOpacity="1"
                    strokeWidth="1"
                    transform="matrix(1.4872568,0,0,1.4575117,-3.9272774,-6.4278868)"
                  />
                </svg>
              </button>
              <button
                className="admin-icons btn btn-default"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                <svg
                  aria-hidden="true"
                  height="35px"
                  role="img"
                  version="1.1"
                  viewBox="0 0 20 20"
                  width="35px"
                  x="0px"
                  xmlSpace="preserve"
                  y="0px"
                >
                  <path
                    className="outer"
                    d="m 15.59322,11.271187 a 6.2288136,6.3559322 0 1 1 -12.4576271,0 6.2288136,6.3559322 0 1 1 12.4576271,0 z"
                    fill="none"
                    id="path2999"
                    stroke="#771122"
                    strokeDasharray="none"
                    strokeMiterlimit="4"
                    strokeOpacity="1"
                    strokeWidth="1"
                    transform="matrix(1.4872568,0,0,1.4575117,-3.9272774,-6.4278868)"
                  />
                </svg>
              </button>
              <button
                className="admin-icons btn btn-default"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                <svg
                  aria-hidden="true"
                  height="35px"
                  role="img"
                  version="1.1"
                  viewBox="0 0 20 20"
                  width="35px"
                  x="0px"
                  xmlSpace="preserve"
                  y="0px"
                >
                  <path
                    className="outer"
                    d="m 15.59322,11.271187 a 6.2288136,6.3559322 0 1 1 -12.4576271,0 6.2288136,6.3559322 0 1 1 12.4576271,0 z"
                    fill="none"
                    id="path2999"
                    stroke="#771122"
                    strokeDasharray="none"
                    strokeMiterlimit="4"
                    strokeOpacity="1"
                    strokeWidth="1"
                    transform="matrix(1.4872568,0,0,1.4575117,-3.9272774,-6.4278868)"
                  />
                </svg>
              </button>
              <button
                className="admin-icons btn btn-default"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                <svg
                  aria-hidden="true"
                  height="35px"
                  role="img"
                  version="1.1"
                  viewBox="0 0 20 20"
                  width="35px"
                  x="0px"
                  xmlSpace="preserve"
                  y="0px"
                >
                  <path
                    className="outer"
                    d="m 15.59322,11.271187 a 6.2288136,6.3559322 0 1 1 -12.4576271,0 6.2288136,6.3559322 0 1 1 12.4576271,0 z"
                    fill="none"
                    id="path2999"
                    stroke="#771122"
                    strokeDasharray="none"
                    strokeMiterlimit="4"
                    strokeOpacity="1"
                    strokeWidth="1"
                    transform="matrix(1.4872568,0,0,1.4575117,-3.9272774,-6.4278868)"
                  />
                </svg>
              </button>
              <button
                className="btn btn-default"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseOut={[Function]}
                onMouseOver={[Function]}
                type="button"
              >
                <span
                  className="assembl-icon-delete grey"
                />
              </button>
            </div>
          </div>
        </div>
        <div
          className="gauge-vote-for-proposal"
        >
          <p>
            I'll generate the mobile SMS bus, that should microchip the TCP matrix!
          </p>
          <div
            className="rc-slider"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onMouseUp={[Function]}
            onTouchStart={[Function]}
            style={undefined}
          >
            <div
              className="rc-slider-rail"
              style={Object {}}
            />
            <div
              className="rc-slider-track"
              style={
                Object {
                  "left": "0%",
                  "visibility": "hidden",
                  "width": "NaN%",
                }
              }
            />
            <div
              className="rc-slider-step"
            />
            <div
              className="rc-slider-handle"
              role="slider"
              style={
                Object {
                  "left": "NaN%",
                }
              }
              tabIndex={0}
            />
            <div
              className="rc-slider-mark"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
